embassy-stm32/src/gpio.rs

```diff
@@ -99,6 +131,7 @@ impl<'d, T: Pin> Output<'d, T> {
             let n = pin.pin() as usize;
             r.pupdr().modify(|w| w.set_pupdr(n, vals::Pupdr::FLOATING));
             r.moder().modify(|w| w.set_moder(n, vals::Moder::OUTPUT));
+            pin.set_speed(speed);
```

Comment:
I think this should set OTYPER, otherwise it'll stay as opendrain if it was used before.

Reply:
Agreed, and done!